msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/scheduler/PrivateMessageSchedulerServiceImpl.java

```diff
@@ -0,0 +1,672 @@
+import org.sakaiproject.api.app.messageforums.MessageForumsMessageManager;
+import org.sakaiproject.user.api.User;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.user.api.UserNotDefinedException;
+import org.sakaiproject.util.ResourceLoader;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class PrivateMessageSchedulerServiceImpl implements PrivateMessageSchedulerService {
+
+	@Setter
+	private PrivateMessageManager prtMsgManager;
+	@Setter
+	private ScheduledInvocationManager scheduledInvocationManager;
+	@Setter
+	private SynopticMsgcntrManager synopticMsgcntrManager;
+	@Setter
+	private EventTrackingService eventTrackingService;
+	@Setter
+	private MessageForumsMessageManager messageManager;
+	@Setter
+	private UserDirectoryService userDirectoryService;
+	@Setter
+	private AuthzGroupService authzGroupService;
+	@Setter
+	private SiteService siteService;
+	@Setter
+	private ServerConfigurationService serverConfigurationService;
+	@Setter
+	private PrivacyManager privacyManager;
+
+	private static final String MESSAGECENTER_BUNDLE = "org.sakaiproject.api.app.messagecenter.bundle.Messages";
+	private ResourceLoader rl;
+
+	public void init() {
+		log.debug("PrivateMessageSchedulerService init()");
+		rl = new ResourceLoader(MESSAGECENTER_BUNDLE);
+	}
+
+	public void destroy() {
+		log.debug("PrivateMessageSchedulerService destroy()");
+	}
+
+	@Override
+	public void scheduleDueDateReminder(Long messageId) {
+		removeScheduledReminder(messageId);
+
+		PrivateMessage pvtMsg = (PrivateMessage) prtMsgManager.getMessageById(messageId);
+		Instant reminderDate = pvtMsg.getScheduledDate().toInstant();
+		scheduledInvocationManager.createDelayedInvocation(reminderDate,
+				"org.sakaiproject.api.app.messageforums.scheduler.PrivateMessageSchedulerService",
+				messageId.toString());
+	}
+
+	@Override
+	public void removeScheduledReminder(Long messageId) {
+		if (messageId != null)
+			scheduledInvocationManager.deleteDelayedInvocation(
+					"org.sakaiproject.api.app.messageforums.scheduler.PrivateMessageSchedulerService",
+					messageId.toString());
+	}
+
+	@Override
+	public void execute(String opaqueContext) {
+		Long messageId = Long.parseLong(opaqueContext);
+		PrivateMessage pvtMsg = (PrivateMessage) prtMsgManager.getMessageById(messageId);
+
+		Map<User, Boolean> recipients = getRecipients(pvtMsg);
+
+		pvtMsg.setCreated(pvtMsg.getScheduledDate());
+		pvtMsg.setModified(pvtMsg.getScheduledDate());
+		pvtMsg.setScheduledDate(null);
+		pvtMsg.setScheduler(false);
+		pvtMsg.setDraft(false);
+
+		prtMsgManager.sendPrivateMessage(pvtMsg, recipients, false);
```

Comment:
why is the sendingEmail always false? what if the user selected to send an email in the future?

Reply:
I don't remember right now why it was marked as false and Victor is currently on paternity leave, he'll be back in a month or so, so any ideas you have to change this behaviour are valid, have you had any bugs with this?

Reply:
@jesusmmp picked up this bug. thanks to Jesus. but the bug is clear: no email is sent even when the instructor wants and expects email to be sent